src/simd/lines_bwd.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::ptr;
+
+use crate::helpers::CoordType;
+
+/// Starting from the `offset` in `haystack` with a current line index of
+/// `line`, this seeks backwards to the `line_stop`-nth line and returns the
+/// new offset and the line index at that point.
+///
+/// Note that this function differs from `lines_fwd` in that it
+/// seeks backwards even if the `line` is already at `line_stop`.
+/// This allows you to ensure (or test) whether `offset` is at a line start.
+///
+/// It returns an offset *past* the newline (= at the start of the next line).
```

Comment:
next -> previous?


```suggestion
/// It returns an offset *past* the newline (= at the start of the prior line).
```

or perhaps I misunderstand

Reply:
It's more like "the current line". I changed the wording a bit to be clearer about this.